File: spring-web-java/src/main/java/spring/web/java/domain/token/Token.java
package spring.web.java.domain.token;

import lombok.Getter;
import org.springframework.data.annotation.Id;
import org.springframework.data.redis.core.RedisHash;

@Getter
@RedisHash(value = "token", timeToLive = 604800)
public class Token {

    @Id
    private Long memberId;
    private String refreshToken;

    public static Token create(Long memberId, String refreshToken) {
        Token token = new Token();

        token.memberId = memberId;
        token.refreshToken = refreshToken;

        return token;
    }
}

File: spring-web-kotlin/src/main/kotlin/spring/web/kotlin/domain/token/Token.kt
package spring.web.kotlin.domain.token

import org.springframework.data.annotation.Id
import org.springframework.data.redis.core.RedisHash

@RedisHash(value = "token", timeToLive = 604800)
class Token protected constructor(
    @Id
    val memberId: Long,
    val refreshToken: String
){
    companion object {
        fun create(memberId: Long, refreshToken: String) {
            Token(
                memberId = memberId,
                refreshToken = refreshToken
            )
        }
    }
}